universal-application-tool-0.0.1/app/auth/saml/SamlCiviFormProfileAdapter.java

```diff
@@ -165,11 +165,12 @@ public CiviFormProfileData mergeCiviFormProfile(
 
   private String extractAttributeFromArrayList(SAML2Profile profile, String attr) {
     ArrayList attributeArray = profile.getAttribute(attr, ArrayList.class);
+    if (attributeArray == null) {
```

Comment:
I think we should prefer Optional.ofNullable(.....) and isPresent checks in general.

Reply:
Update: Blocking: please use optional.

I discussed with Bion and going forward our code should prefer Optional over null .